models/Menus_model.php

```diff
@@ -15,13 +15,7 @@ public function getMorphClass()
 
     public function isAvailable()
     {
-        if (!$mealtime = $this->mealtime)
-            return TRUE;
-
-        if (!$mealtime->mealtime_status)
-            return TRUE;
-
-        return $mealtime->isAvailableNow();
+        return $this->isAvailable();
     }
```

Comment:
Maybe have this method check if the menu has mealtimes and is available for the current time, instead of calling itself? Basically the same as the changed behaviour but on multiple mealtimes.